pkg/controller/seed-controller-manager/kubernetes/health.go

```diff
@@ -157,6 +166,11 @@ func (r *Reconciler) machineControllerHealthCheck(ctx context.Context, cluster *
 	}
 }
 
+func (r *Reconciler) applicationControllerHealthCheck(ctx context.Context, cluster *kubermaticv1.Cluster, namespace string) (kubermaticv1.HealthStatus, error) {
+	// TODO: Implement this
```

Comment:
I have added the skeleton code for checking the health of application-controller and integrated it everywhere. This can be implemented later when we have the required pieces in place i.e. deployments, webhooks, CRDs, etc.

Reply:
This has been implemented now